scripts/nbconverted/high_weight_genes.py

```diff
@@ -0,0 +1,355 @@
+
+
+# ## Visualize distributions
+# 
+# Certain `high_weight_genes` methods enable quick visualizations of high weigh gene features
+# 
+# 1. Positive and negative tail number of high weight genes
+# 2. Full gene activity distributions
+# 
+# ### Positive/negative tails
+
+# In[7]:
+
+pca_weights.count_high_weight_genes(return_plot=True, title='PCA',
+                                    color=algorithm_color_dict['PCA']);
+
+
+# In[8]:
+
+ica_weights.count_high_weight_genes(return_plot=True, title='ICA',
+                                    color=algorithm_color_dict['ICA']);
+
+
+# In[9]:
+
+nmf_weights.count_high_weight_genes(return_plot=True, title='NMF',
+                                    color=algorithm_color_dict['NMF']);
+
+
+# In[10]:
+
+adage_weights.count_high_weight_genes(return_plot=True, title='ADAGE',
+                                      color=algorithm_color_dict['ADAGE']);
+
+
+# In[11]:
+
+tybalt_weights.count_high_weight_genes(return_plot=True, title='Tybalt',
+                                       color=algorithm_color_dict['Tybalt']);
+
+
+# In[12]:
+
+vae_2l100_weights.count_high_weight_genes(return_plot=True,
+                                          title='VAE 2 Hidden Layer (100)',
+                                          color=algorithm_color_dict['VAE_100']);
+
+
+# In[13]:
+
+vae_2l300_weights.count_high_weight_genes(return_plot=True,
+                                          title='VAE 2 Hidden Layer (300)',
+                                          color=algorithm_color_dict['VAE_300']);
+
+
+# ### Full gene activity distributions
+# 
+# Plot the gene activity distributions of the first 10 features for each algorithm
+
+# In[14]:
+
+first_encodings = range(1, 11)
+pca_weights.plot_weight_dist(features=first_encodings, title='PCA',
+                             color=algorithm_color_dict['PCA']);
+
+
+# In[15]:
+
+ica_weights.plot_weight_dist(features=first_encodings, title='ICA',
+                             color=algorithm_color_dict['ICA']);
+
+
+# In[16]:
+
+nmf_weights.plot_weight_dist(features=first_encodings, title='NMF',
+                             color=algorithm_color_dict['NMF']);
+
+
+# In[17]:
+
+adage_weights.plot_weight_dist(features=first_encodings, title='ADAGE',
+                               color=algorithm_color_dict['ADAGE']);
+
+
+# In[18]:
+
+tybalt_weights.plot_weight_dist(features=first_encodings, title='Tybalt',
+                                color=algorithm_color_dict['Tybalt']);
+
+
+# In[19]:
+
+vae_2l100_weights.plot_weight_dist(features=first_encodings,
+                                   title='VAE 2 Hidden Layer (100)',
+                                   color=algorithm_color_dict['VAE_100']);
+
+
+# In[20]:
+
+vae_2l300_weights.plot_weight_dist(features=first_encodings,
+                                   title='VAE 2 Hidden Layer (300)',
+                                   color=algorithm_color_dict['VAE_300']);
+
+
+# ## Observe distribution of skewness and kurtosis
```

Comment:
Add note about the outliers here?

Reply:
Yep - will do. They are interesting in their own right. I think treating them differently from other features will be important. In this dataset, they all seem to be finding similar things. I will need to describe them in more detail at some point (see #96).